modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/reader/BatchEngineImportTaskItemReaderUtil.java

```diff
@@ -146,11 +153,26 @@ public static <T> T convertValue(
 			}
 		}
 
-		for (ItemReaderPostAction itemReaderPostAction :
-				itemReaderPostActions) {
+		try {
+			for (ItemReaderPostAction itemReaderPostAction :
+					itemReaderPostActions) {
+
+				itemReaderPostAction.run(
+					batchEngineImportTask, extendedProperties, item);
+			}
+		}
+		catch (ReflectiveOperationException reflectiveOperationException) {
+			Throwable throwable = reflectiveOperationException.getCause();
+
+			if (throwable instanceof ValidationException validationException) {
+				batchEngineImportTaskExceptionHandlers.forEach(
+					batchEngineImportTaskExceptionHandler ->
+						batchEngineImportTaskExceptionHandler.handle(
```

Comment:
Hey @peerkar ideally we should have just one place calling these handlers... Maybe there is a higher layer in which we can move both of the 2 places we have now. What do you think?